app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/StatisticsModule.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.android.architecture.blueprints.todoapp.di
+
+import androidx.lifecycle.ViewModel
+import com.example.android.architecture.blueprints.todoapp.statistics.StatisticsFragment
+import com.example.android.architecture.blueprints.todoapp.statistics.StatisticsUtils
+import com.example.android.architecture.blueprints.todoapp.statistics.StatisticsViewModel
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.android.ContributesAndroidInjector
+import dagger.multibindings.IntoMap
+
+/**
+ * Dagger module for the Statistics feature.
+ *
+ * It shows how to include a module that contains @Provides methods.
+ */
+@Module(includes = [StatsUtilsModule::class])
+abstract class StatisticsModule {
+
+    @ContributesAndroidInjector(modules = [
+        ViewModelBuilder::class
+    ])
+    internal abstract fun statisticsFragment(): StatisticsFragment
+
+    @Binds
+    @IntoMap
+    @ViewModelKey(StatisticsViewModel::class)
+    abstract fun bindViewModel(viewmodel: StatisticsViewModel): ViewModel
+
+}
+
+@Module
+class StatsUtilsModule {
+
+    @Provides
+    fun provideStatsUtil() = StatisticsUtils()
```

Comment:
Move to constructor injection?

Reply:
Isn't it better to keep classes dagger-agnostic and have all DI-related stuff in modules and components?

Reply:
Yes but inject annotation is not dagger specific. It is part of jsr for inject. Really you should be decoupling your consumers from producers. Only inject interfaces into constructors which are  bindings from an impl that you declare in a dagger module. That is the cleanest approach, no dagger in impls and no coupling between implementations

Reply:
Ok, PTAL!